public/app/features/provisioning/Wizard/FinishStep.tsx

```diff
@@ -0,0 +1,141 @@
+import { useEffect } from 'react';
+import { Controller, useFormContext } from 'react-hook-form';
+import { useNavigate } from 'react-router-dom-v5-compat';
+
+import { Alert, Field, FieldSet, Input, MultiCombobox, Stack, Switch, Text } from '@grafana/ui';
+
+import { getWorkflowOptions } from '../ConfigForm';
+import { checkPublicAccess, checkImageRenderer } from '../GettingStarted/features';
+import { GETTING_STARTED_URL } from '../constants';
+
+import { WizardFormData } from './types';
+
+export function FinishStep() {
+  const {
+    register,
+    watch,
+    control,
+    formState: { errors },
+  } = useFormContext<WizardFormData>();
+
+  const type = watch('repository.type');
+  const isGithub = type === 'github';
+  const isPublic = checkPublicAccess();
+  const hasImageRenderer = checkImageRenderer();
+  // Enable sync by default
+  const { setValue } = useFormContext<WizardFormData>();
+  const navigate = useNavigate();
+
+  // Set sync enabled by default
+  useEffect(() => {
+    setValue('repository.sync.enabled', true);
```

Comment:
This needs to be done in the form's default values, whenever you need to set the form state like this, it's most likely you're doing smth wrong (apart from a minor set of valid cases).

Reply:
It's on purpose. We start with false and on the last step we implicitly set it to true so that the automatic pulling starts once the initial pull or migrate happened.

We could default to false in the defaults and then enable in the last step by setting the form value. Is that what you mean?